plugins/de.cognicrypt.codegenerator/src/main/java/de/cognicrypt/codegenerator/generator/CrySLBasedCodeGenerator.java

```diff
@@ -1232,6 +1232,7 @@ public GeneratorClass setUpTemplateClass(String pathToTemplateFile) {
 			Map<SimpleName, CrySLObject> postCGVars = new HashMap<SimpleName, CrySLObject>();
 
 			List<CodeGenCrySLRule> rules = new ArrayList<CodeGenCrySLRule>();
+			List<CrySLRule> rulesFromSootRunner = SootRunner.getRules(getDeveloperProject().project);

```

Comment:
Why did you move this up?

Reply:
Because the `visit` method is called multiple times, the rules are retrieved repeatedly if `SootRunner.getRules` is inside this method.